subsys/bluetooth/mesh/cfg_cli.c

```diff
@@ -2443,3 +2443,36 @@ struct bt_mesh_comp_p1_ext_item *bt_mesh_comp_p1_pull_ext_item(
 	}
 	return ext_item;
 }
+
+struct bt_mesh_comp_p2_record *bt_mesh_comp_p2_record_pull(struct net_buf_simple *buf,
+							 struct bt_mesh_comp_p2_record *elem)
+{
+	if (buf->len < 6) {
```

Comment:
Should it be 8? The minimum possible record length is 8 assuming that `Num_Element_Offsets` and `Additional_Data_Len` are zero (though not sure if it's allowed to have `Num_Element_Offsets` zero, but this is another story.

Reply:
I think it is covered by the next check. We pull the six first bytes and then do this error check: `if (buf->len < elem->elem_offset_cnt + 2)`

Reply:
But when I think about it we might as well exit early if it is not 8.